### DISCUSSION TITLE
uPnP not working for 3.0 but works for 2.94

### DISCUSSION THREAD

**Discussion by ieee8023** [transmission/transmission#8487] [Support]
I just upgraded to Transmission 3.0 and the uPnP stopped working. If I open 2.94 I can see that a uPnP entry is made. My router is using  DD-WRT and other torrent clients are working fine.

I'm using Mac OSX 10.13

I see this error in the logs for v3.0
```
natpmp.c:73 [Info] Port Forwarding (NAT-PMP): initnatpmp succeeded (0)
natpmp.c:73 [Info] Port Forwarding (NAT-PMP): sendpublicaddressrequest succeeded (2)
natpmp.c:78 [Debug] Port Forwarding (NAT-PMP): readnatpmpresponseorretry failed. Natpmp returned -7 (the gateway does not support nat-pmp); errno is 61 (Connection refused)
upnp.c:208 [Debug] Port Forwarding (UPnP): UPNP_GetValidIGD failed (errno 0 - Undefined error: 0)
upnp.c:209 [Debug] Port Forwarding (UPnP): If your router supports UPnP, please make sure UPnP is enabled!
```
For v2.94 I see this (with IPs replaced):
```
natpmp.c:70 [Info] Port Forwarding (NAT-PMP): initnatpmp succeeded (0)
upnp.c:304 [Info] Port Forwarding (UPnP): Port forwarding through "http://{gateway address}:1780/control?WANIPConnection", service "urn:schemas-upnp-org:service:WANIPConnection:1". (local address: {host address}:55556)
upnp.c:227 [Info] Port Forwarding (UPnP): Found Internet Gateway Device "http://{gateway address}:1780/control?WANIPConnection"
```
If I manually configure port forwarding while using v3.0 it works.

**Comment by calavera12** [comment]
Same issue.  UPnP works with 2.94 (random port, open, green dot), but DOES not work with 3.00 (random port, closed, red dot)

**Comment by tangestani** [comment]
Same issue here. uPnP works on 2.94, not in 3.00.

<img width="1070" alt="Screen Shot 2020-05-31 at 11 16 23 AM" src="https://user-images.githubusercontent.com/548320/83348058-0e30e880-a332-11ea-83cd-c8c745cddfa8.png">

The fix for #347 seems to have caused this issue. Downgrading miniupnpc to version 1.9 fixes the issue.

<img width="909" alt="Screen Shot 2020-05-31 at 12 12 01 PM" src="https://user-images.githubusercontent.com/548320/83348850-c1044500-a338-11ea-953d-90b901351be2.png">

FWIW the latest release of minipnpc (v2.1) has the same issue.

**Comment by 0lm** [comment]
kinda sucks that upnp is not working on 3.00. and i wondered why my downloads werent starting until i saw this post. hope it will be fixed soon

**Comment by CoelacanthusHex** [comment]
Same issue here.

**Comment by ghost** [comment]
Same issue here.

**Comment by DanAykroyd256** [comment]
Same. Any updates?

**Comment by Coockie** [comment]
I have two computers connected to ZTE router. One connected by ethernet cable, another - by Wi-Fi dongle. I have working UPnP in 3.0 and in 43b9d5c147 but not working in 5efec26a3b on machine connected by ethernet cable. On second computer i haven't working UPnP since 3.0 version. By not working UPnP i mean that transmission cannot open port.

**Comment by Coockie** [comment]
I retested early version of transmission such as 2.94 and 2.91 and these versions of transmission cannot open port on computer with Wi-Fi dongle. Also i retested 3.00 and 3.00+ e908193027 transmission version and here i found out that in my router's settings transmission 3.00 create wrong UPnP entries see screenshot. My local IP on this machine is 192.168.0.5
![ZTE_RT-GM-5_Transmission_300](https://user-images.githubusercontent.com/6870054/152111960-d6d3d90c-7815-49a2-9027-d590897e158b.jpg)
But 3.00+ e908193027 didn't create anything related. Also i gathered some wireshark's dumps when launch and exit the application.
[Transmission Wireshark.zip](https://github.com/transmission/transmission/files/7984417/Transmission.Wireshark.zip)

Qbittorrent on this computer successfull open ports.

**Comment by zaid4412100** [comment]
Can confirm in Debian (Stable and testing). Works with another torrent client.

**Comment by iliqiliev** [comment]
I also have the same / nearly the same ZTE router and it UPnP works fine with qBittorent but not with transmission-{gtk,cli} 4.0.6-2

**Comment by Rogue-Git-Dev** [comment]
I can confirm using the latest trunk build of the Mac client UPnP does not work:
> port-forwarding-natpmp.cc:52: readnatpmpresponseorretry failed. Natpmp returned -7 (the gateway does not support nat-pmp); errno is 61 (Connection refused)

I use three Eero Max 7 nodes now but even previous to that I always had UPnP enabled and had IPV4 and IPV6 reservations with rules for my Mac's LAN IP addresses to forward ports 49152—65535. I doubt it works however due to the aforementioned message ad infinitum from the log. I really know not what else to do to troubleshoot it. Nothing is blocked in NextDNS; I made sure of that long ago. Perhaps I must contact Eero and ask them why this is as such.

**Comment by dreua** [comment]
This is a mac specific issue, correct? Maybe someone could put that in the title or a label for better overview. On Fedora Linux I had some issues in the meantime (and subscribed to the wrong issue it seems) but now with 4.0 it all works perfectly fine.

**Comment by Rogue-Git-Dev** [comment]
Having been writ by David Auer on Monday 17 March 2025 at 05:18:48 -0500 ⬇️

> This is a mac specific issue, correct?
Yes. I just responded to an issue already open.

-- 
Blind Barista Sir J.

**Comment by zaid4412100** [comment]
> This is a mac specific issue, correct? Maybe someone could put that in the title or a label for better overview. On Fedora Linux I had some issues in the meantime (and subscribed to the wrong issue it seems) but now with 4.0 it all works perfectly fine.

@dreua  no it isn't as I am having the same issue on Debian Testing.

**Comment by zaid4412100** [comment]
Maybe the issue is CGNAT?
